spec/sec-regexp-regular-expression-objects-patch.html

```diff
@@ -104,343 +104,411 @@ <h1>Runtime Semantics: BackreferenceMatcher ( _n_, _direction_ )</h1>
-      1. Return ! CreateRegExpStringIterator(_matcher_, _S_, _global_, _fullUnicode_<ins>, _mapfn_</ins>).
-    </emu-alg>
+    <emu-clause id="sec-createregexpmatchiterator" aoid="CreateRegExpMatchIterator">
+      <h1><del>CreateRegExpStringIterator</del><ins>CreateRegExpMatchIterator</ins> ( _R_, _S_, _global_, _fullUnicode_<ins>, _matchOptions_</ins> )</h1>
+      <p>The abstract operation <del>_CreateRegExpStringIterator_</del><ins>_CreateRegExpMatchIterator_</ins> is used to create such iterator objects. It performs the following steps:</p>
+      <emu-alg>
+        1. Assert: Type(_S_) is String.
+        1. Assert: Type(_global_) is Boolean.
+        1. Assert: Type(_fullUnicode_) is Boolean.
+        1. <ins>Assert: _matchOptions_ is MatchOptions Record.</ins>
+        1. Let _iterator_ be ObjectCreate(<emu-xref href="#table-properties-of-regexp-match-iterator-instances">%RegExp<del>String</del><ins>Match</ins>IteratorPrototype%</emu-xref>, &laquo; [[IteratingRegExp]], [[IteratedString]], [[Global]], [[Unicode]]<ins>, [[MatchOptions]]</ins>, [[Done]] &raquo;).
+        1. Set _iterator_.[[IteratingRegExp]] to _R_.
+        1. Set _iterator_.[[IteratedString]] to _S_.
+        1. Set _iterator_.[[Global]] to _global_.
+        1. Set _iterator_.[[Unicode]] to _fullUnicode_.
+        1. <ins>Set _iterator_.[[MatchOptions]] to _matchOptions_.</ins>
+        1. Set _iterator_.[[Done]] to *false*.
+        1. Return _iterator_.
+      </emu-alg>
+    </emu-clause>
   </emu-clause>
 
-  <emu-clause id="sec-createregexpstringiterator" aoid="CreateRegExpStringIterator">
-    <h1>CreateRegExpStringIterator ( _R_, _S_, _global_, _fullUnicode_<ins>, _mapfn_</ins> )</h1>
-    <emu-note>This is part of the <a href="https://github.com/tc39/proposal-string-matchall">String.prototype.matchAll</a> proposal.</emu-note>
+</emu-clause>
 
-    <p>The abstract operation _CreateRegExpStringIterator_ is used to create such iterator objects. It performs the following steps:</p>
-    <emu-alg>
-      1. Assert: Type(_S_) is String.
-      1. Assert: Type(_global_) is Boolean.
-      1. Assert: Type(_fullUnicode_) is Boolean.
-      1. <ins>Assert: _mapfn_ is *undefined* or IsCallable(_mapfn_) is *true*.</ins>
-      1. Let _iterator_ be ObjectCreate(<emu-xref href="#%RegExpStringIteratorPrototype%">%RegExpStringIteratorPrototype%</emu-xref>, &laquo; [[IteratingRegExp]], [[IteratedString]], [[Global]], [[Unicode]]<ins>, [[Mapper]], [[OffsetMapper]]</ins>, [[Done]] &raquo;).
-      1. Set _iterator_.[[IteratingRegExp]] to _R_.
-      1. Set _iterator_.[[IteratedString]] to _S_.
-      1. Set _iterator_.[[Global]] to _global_.
-      1. Set _iterator_.[[Unicode]] to _fullUnicode_.
-      1. <ins>Set _iterator_.[[Mapper]] to _mapfn_.</ins>
-      1. Set _iterator_.[[Done]] to *false*.
-      1. Return _iterator_.
-    </emu-alg>
-  </emu-clause>
+<emu-clause id="sec-regexp-match-iterator-objects">
+  <h1>RegExp <del>String</del><ins>Match</ins> Iterator Objects</h1>
```

Comment:
they’re still iterating a string - it doesn’t matter what they produce.
```suggestion
  <h1>RegExp String Iterator Objects</h1>
```